[cas-484] Thread replies footnote for different message types

samiuelson opened the pull request:
https://stream-io.atlassian.net/browse/CAS-484

### Description

Adding thread replies count label for all of the available message types. 

<img width="456" alt="Screenshot 2020-12-31 at 12 39 36" src="https://user-images.githubusercontent.com/4527432/103409389-deb3ba00-4b66-11eb-9900-5ffb2dfc0cbe.png">
<img width="454" alt="Screenshot 2020-12-31 at 12 39 25" src="https://user-images.githubusercontent.com/4527432/103409393-dfe4e700-4b66-11eb-8d86-c6fb1f3782c6.png">
<img width="455" alt="Screenshot 2020-12-31 at 12 38 54" src="https://user-images.githubusercontent.com/4527432/103409395-e07d7d80-4b66-11eb-9083-3a128d788e6f.png">
<img width="458" alt="Screenshot 2020-12-31 at 12 38 29" src="https://user-images.githubusercontent.com/4527432/103409396-e1161400-4b66-11eb-8c18-0d899fbf02fe.png">


### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Changelog updated with client-facing changes
~- [ ] New code is covered by unit tests~
- [x] Comparison screenshots added for visual changes
- [x] Reviewers added

zsmb13 commented:
On the design, the timestamps and the thread reply info are at the same height and are the same font size. Looking at the screenshots here, I think we're off by a bit on one of them.

![image](https://user-images.githubusercontent.com/12054216/103353013-fe37dd80-4aa7-11eb-8c30-31a89fd8a07a.png)

zsmb13 commented on stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/viewholder/decorator/ThreadRepliesDecorator.kt:
```diff
@@ -17,28 +23,187 @@ internal class ThreadRepliesDecorator : BaseDecorator() {
+        setupThreadRepliesView(viewHolder.binding, data)
+    }
+
+    private fun setupThreadRepliesView(
+        binding: StreamUiItemMessagePlainTextWithFileAttachmentsBinding,
+        data: MessageListItem.MessageItem
```
Looks like we're duplicating a lot of code here in this decorator. Instead of passing in the binding to these methods, you could have a single method and pass in the Views to manipulate (the root, and then the `threadRepliesFootnote`) as two parameters to it.

Take a look at some of the other decorators, like `ReplyDecorator` or `MessageFooterDecorator` for this pattern.

samiuelson replied:
I've made the `setupThreadRepliesView()` reusable. Let me know if you can see a better way of doing this.

samiuelson commented:
> On the design, the timestamps and the thread reply info are at the same height and are the same font size. Looking at the screenshots here, I think we're off by a bit on one of them.
> 
> ![image](https://user-images.githubusercontent.com/12054216/103353013-fe37dd80-4aa7-11eb-8c30-31a89fd8a07a.png)

Cleaned up text sizes, dimens, and alignment of views.
Got rid of `android:fontWeight="600"`, replaced by `android:textStyle="bold"` to be compatible with all supporte OS versions.

leandroBorgesFerreira reviewed (approved):
looks good to me
